packages/inference/src/providers/wavespeed-ai.ts

```diff
@@ -0,0 +1,193 @@
+		if (!isBinary) {
+			headers["Content-Type"] = "application/json";
+		}
+		return headers;
+	}
+
+	override async getResponse(
+		response: WaveSpeedAIResponse<WaveSpeedAISubmitResponse>,
+		url?: string,
+		headers?: Record<string, string>
+	): Promise<Blob> {
+		if (!headers && this.accessToken) {
+			headers = { Authorization: `Bearer ${this.accessToken}` };
+		}
+		if (!headers) {
+			throw new InferenceOutputError("Headers are required for WaveSpeed AI API calls");
+		}
+
+		const resultUrl = response.data.urls.get;
+
+		// Poll for results until completion
+		while (true) {
+			const resultResponse = await fetch(resultUrl, { headers });
+
+			if (!resultResponse.ok) {
+				throw new InferenceOutputError(`Failed to get result: ${resultResponse.statusText}`);
+			}
+
+			const result: WaveSpeedAIResponse = await resultResponse.json();
+			if (result.code !== 200) {
+				throw new InferenceOutputError(`API request failed with code ${result.code}: ${result.message}`);
+			}
+
+			const taskResult = result.data;
+
+			switch (taskResult.status) {
+				case "completed": {
+					// Get the video data from the first output URL
+					if (!taskResult.outputs?.[0]) {
+						throw new InferenceOutputError("No video URL in completed response");
+					}
+					const videoResponse = await fetch(taskResult.outputs[0]);
+					if (!videoResponse.ok) {
+						throw new InferenceOutputError("Failed to fetch video data");
+					}
+					return await videoResponse.blob();
+				}
+				case "failed": {
+					throw new InferenceOutputError(taskResult.error || "Task failed");
+				}
+				case "processing":
+				case "created":
+					// Wait before polling again
+					await delay(100);
+					continue;
+
+				default: {
+					throw new InferenceOutputError(`Unknown status: ${taskResult.status}`);
+				}
+			}
+		}
+	}
+}
+
+export class WavespeedAITextToImageTask extends WavespeedAITask implements TextToImageTaskHelper {
+	constructor() {
+		super(WAVESPEEDAI_API_BASE_URL);
+	}
+}
+
+export class WavespeedAITextToVideoTask extends WavespeedAITask implements TextToVideoTaskHelper {
+	constructor() {
+		super(WAVESPEEDAI_API_BASE_URL);
+	}
+}
+
+export class WavespeedAIImageToImageTask extends WavespeedAITask implements ImageToImageTaskHelper {
+	constructor() {
+		super(WAVESPEEDAI_API_BASE_URL);
+	}
+
+	async preparePayloadAsync(args: ImageToImageArgs): Promise<RequestArgs> {
+		if (!args.parameters) {
+			return {
+				...args,
+				model: args.model,
+				data: args.inputs,
+			};
+		} else {
+			return {
+				...args,
+				inputs: base64FromBytes(
+					new Uint8Array(args.inputs instanceof ArrayBuffer ? args.inputs : await (args.inputs as Blob).arrayBuffer())
+				),
+			};
+		}
+	}
+
+	override preparePayload(params: BodyParams): Record<string, unknown> {
+		return {
+			...omit(params.args, ["inputs", "parameters"]),
+			...(params.args.parameters as Record<string, unknown>),
+			image: params.args.inputs,
+		};
+	}
```

Comment:
I think only one of the two ( `preparePayload` or `preparePayloadAsync`) should be responsible for building the payload, meaning, I'd rather move the rename of `inputs` to `image` in `preparePayloadAsync` an have `preparePayload` as dumb as possible

cc @hanouticelina - would love your opinion on that specific point

Reply:
I only kept preparePayloadAsync func